core/src/main/java/org/jboss/jandex/IndexView.java

```diff
@@ -153,17 +153,107 @@ default Collection<ClassInfo> getAllKnownSubclasses(final Class<?> clazz) {
     }
 
     /**
-     * Gets all known direct implementors of the specified interface name. A known
+     * Gets all known direct subinterfaces of the specified interface. A known direct
+     * subinterface is one which was found during the scanning process; however, this is
+     * often not the complete universe of subinterfaces, since typically indexes are
+     * constructed per jar. It is expected that several indexes will need to be searched
+     * when analyzing a jar that is a part of a complex multi-module/classloader
+     * environment (like an EE application server).
+     * <p>
+     * Note that this will only pick up direct subinterfaces of the interface. It will not
+     * pick up subinterfaces of subinterfaces.
+     *
+     * @param interfaceName the super interface of the desired subinterfaces
+     * @return a non-null list of all known subinterfaces of interfaceName
+     * @since 3.0
+     */
+    Collection<ClassInfo> getKnownDirectSubinterfaces(DotName interfaceName);
+
+    /**
+     * Gets all known direct subinterfaces of the specified interface. A known direct
+     * subinterface is one which was found during the scanning process; however, this is
+     * often not the complete universe of subinterfaces, since typically indexes are
+     * constructed per jar. It is expected that several indexes will need to be searched
+     * when analyzing a jar that is a part of a complex multi-module/classloader
+     * environment (like an EE application server).
+     * <p>
+     * Note that this will only pick up direct subinterfaces of the interface. It will not
+     * pick up subinterfaces of subinterfaces.
+     *
+     * @param interfaceName the super interface of the desired subinterfaces
+     * @return a non-null list of all known subinterfaces of interfaceName
+     * @since 3.0
+     */
+    default Collection<ClassInfo> getKnownDirectSubinterfaces(String interfaceName) {
```

Comment:
Are these versions of the method really needed? I mean it can be useful yes, but all other methods accept `DotName` params so one could ask _"why string and class only for these two methods"_? Also clients may prefer these versions of methods which is suboptiomal because (1) componentized names are preferred and (2) it's always better to share `DotName` instances and avoid unnecessary allocations...

Reply:
All other methods also have a `String` and `Class` overloads since https://github.com/wildfly/jandex/pull/155

When it comes to componentized `DotName`s, I think of it as a two-way street. When storing `DotName`s in a humongous data structure that must be as compact as possible (such as a Jandex index itself), using a componentized form is strongly preferred.

When it comes to one-off usages (such as these methods that lookup data from the index), it doesn't really matter in my opinion. If you share `DotName` instances (typically by storing them into `static final` fields), you prevent unnecessary allocations at the expense of higher old gen usage. If you create them on demand, you cause higher allocation rate. In this particular case, the freshly created `DotName` instances are never stored anywhere, so they die very young, and collecting a new gen is rather cheap.

So I don't really think this is an issue.

Reply:
> All other methods also have a `String` and `Class` overloads since #155

Ok, didn't know about that one.

> 
> When it comes to componentized `DotName`s, I think of it as a two-way street. When storing `DotName`s in a humongous data structure that must be as compact as possible (such as a Jandex index itself), using a componentized form is strongly preferred.

AFAIK the componentized version also has a bit more efficient `equals()`...

> 
> When it comes to one-off usages (such as these methods that lookup data from the index), it doesn't really matter in my opinion. If you share `DotName` instances (typically by storing them into `static final` fields), you prevent unnecessary allocations at the expense of higher old gen usage. If you create them on demand, you cause higher allocation rate. In this particular case, the freshly created `DotName` instances are never stored anywhere, so they die very young, and collecting a new gen is rather cheap.

Ok, I won't argue ;-) although I think that these methods make the API less readable and the _"collecting a new gen is rather cheap"_ argumet is a bit disputable.

Reply:
Actually I had to fix `DotName.equals()` for the case when both `DotName`s are componentized, because the previous implementation assumed that componentization is canonical, which it is not (see #175 and #176 if you wanna know more, but beware, it's a long brain dump of mine :-) ). So `DotName.equals()` is actually most efficient when both `DotName`s are _not_ componentized, because that's just a string comparison.

Performance disputes are meaningless without measurements, and I don't have any -- my only argument is that Stuart clearly thought it's fine in #155 :-)

Reply:
> Actually I had to fix `DotName.equals()` for the case when both `DotName`s are componentized, because the previous implementation assumed that componentization is canonical, which it is not (see #175 and #176 if you wanna know more, but beware, it's a long brain dump of mine :-) ). So `DotName.equals()` is actually most efficient when both `DotName`s are _not_ componentized, because that's just a string comparison.

Interesting! 

> 
> Performance disputes are meaningless without measurements, and I don't have any -- my only argument is that Stuart clearly thought it's fine in #155 :-)

+1 :D